Tabs: documentation page update after API changes (#1622)
iTwin/iTwinUI: apps/website/src/pages/docs/tabs.mdx

  <Tabs.TabDescription>A red fruit</Tabs.TabDescription>
</Tabs.Tab>
```

### Legacy API

Alternatively, you can use Tabs directly with labels, actions, and children as panels. Although this pattern can suffice for simple cases, we strongly recommend using composition API for the best experience.

The following code is equivalent to the [composition example](#composition-api) above:

```tsx
<Tabs
  labels={[
    <Tab key='apple' label='Apple' />,
    <Tab key='orange' label='Orange' />,
    <Tab key='pear' label='Pear' />,
  ]}
>
  {/* … content for active tab */}
</Tabs>
```

### Controlled state

By default, the tabs maintain their own state. You can specify the initially active tab using the `defaultValue` prop in `Tabs.Wrapper`; the tabs will continue to maintain their own state after the initial render.

If you need to manually maintain the active state, you can use `value` and `onValueChange` props in `Tabs.Wrapper`.

<LiveExample src='Tabs.controlled.tsx'>
  <AllExamples.TabsControlledExample client:load />
</LiveExample>

## Variants

### Vertical

<LiveExample src='Tabs.vertical.tsx'>
  <AllExamples.TabsVerticalExample client:load />
</LiveExample>

### Borderless

> 

Reviewer: ```suggestion

The `type` prop can be set to `"borderless"` to show borderless tabs.

```

As their name implies, borderless tabs have no frame or stroke surrounding them.

Use borderless tabs when the tabbed content can be easily told apart from its surroundings (or lack thereof).

<LiveExample src='Tabs.borderless.tsx'>
  <AllExamples.TabsBorderlessExample client:load />
</LiveExample>

### Pill

Pill tabs are meant to be used with content that belong to a similar category. They act like a filter for toggling through different states of similar content.

The maximum recommended amount of pill tabs is 4.

<LiveExample src='Tabs.pill.tsx'>
  <AllExamples.TabsPillExample client:load />
</LiveExample>

### Actions

> You can add action buttons to the end of tab list.

Reviewer: ```suggestion
The `Tabs.Action` subcomponent can be used to add action buttons to the end of tab list.
```

    <Tabs.Tab value='orange' label='Orange' key='orange' />
    <Tabs.Tab value='pear' label='Pear' key='pear' />
  </Tabs.TabList>

  <Tabs.Panel value='apple' key='apple'>
    {/*...Content for apple*/}
  </Tabs.Panel>
  <Tabs.Panel value='orange' key='orange'>
    {/*...Content for orange*/}
  </Tabs.Panel>
  <Tabs.Panel value='pear' key='pear'>
    {/*...Content for pear*/}
  </Tabs.Panel>
</Tabs.Wrapper>
```

`Tabs.Tab` and `Tabs.Panel` can be associated with each other by passing them the same `value`.

`Tabs.Tab` has `label` prop for ease of use. You can also build `Tabs.Tab` with subcomponents and add sublabel and icon to your tab.

```tsx
<Tabs.Tab value='apple' key='apple' label='Apple' />
```

```tsx
<Tabs.Tab value='apple' key='apple'>
  <Tabs.TabIcon>
    <SvgAppleIcon />
  </Tabs.TabIcon>
  <Tabs.TabLabel>Apple</Tabs.TabLabel>
  <Tabs.TabDescription>A red fruit</Tabs.TabDescription>
</Tabs.Tab>
```

### Props API

Alternatively, you can use Tabs directly with labels, actions, and children as panels. Although this pattern can suffice for simple cases, we strongly recommend using composition API for the best experience.

> The following code is equivalent to the composition example above:

Reviewer: lets add a link to the composition api section now that we have it?

```suggestion
The following code is equivalent to the [composition example](#composition-api) above:
```

```tsx
<Tabs
  labels={[
    <Tab key='apple' label='Apple' />,
    <Tab key='orange' label='Orange' />,
    <Tab key='pear' label='Pear' />,
  ]}
>
  {/* … content for active tab */}
</Tabs>
```

### Controlled state

> You can use `value` and `onValueChange` in `Tabs.Wrapper` to have fully controlled state. You are manually controlling active tab in the example bellow.
> 
> You can also change tab that is active on component render using `defaultValue` prop.

Reviewer: ```suggestion
By default, the tabs maintain their own state. You can specify the initially active tab using the `defaultValue` prop in `Tabs.Wrapper`; the tabs will continue to maintain their own state after the initial render.

If you need to manually maintain the active state, you can use `value` and `onValueChange` props in `Tabs.Wrapper`.
```

<LiveExample src='Tabs.controlled.tsx'>
  <AllExamples.TabsControlledExample client:load />
</LiveExample>

## Variants

### Vertical

<LiveExample src='Tabs.vertical.tsx'>
  <AllExamples.TabsVerticalExample client:load />
</LiveExample>

### Borderless

As their name implies, borderless tabs have no frame or stroke surrounding them.

Use borderless tabs when the tabbed content can be easily told apart from its surroundings (or lack thereof).

<LiveExample src='Tabs.borderless.tsx'>
  <AllExamples.TabsBorderlessExample client:load />
</LiveExample>

### Pill

Pill tabs are meant to be used with content that belong to a similar category. They act like a filter for toggling through different states of similar content.

The maximum recommended amount of pill tabs is 4.

<LiveExample src='Tabs.pill.tsx'>
  <AllExamples.TabsPillExample client:load />
</LiveExample>

### Actions

You can add action buttons to the end of tab list.

<LiveExample src='Tabs.actions.tsx'>
  <AllExamples.TabsActionsExample client:load />
</LiveExample>

> ### Overflow

Reviewer: we could remove the overflow section as it does not need any code changes

Author: Removed

import PropsTable from '~/components/PropsTable.astro';
import LiveExample from '~/components/LiveExample.astro';
import * as AllExamples from 'examples';

<p>{frontmatter.description}</p>

<LiveExample src='Tabs.main.tsx'>
  <AllExamples.TabsMainExample client:load />
</LiveExample>

## Usage

### Composition API

You can build Tabs with subcomponents that are fully customizable.

> ```js

Reviewer 2: Should this be tsx?

```suggestion
```tsx
```

Author: Yup, updated

<Tabs.Wrapper>
  <Tabs.TabList>
    <Tabs.Tab value='apple' label='Apple' key='apple' />
    <Tabs.Tab value='orange' label='Orange' key='orange' />

>     <Tabs.Tab value='pear' label='Pear' key='pear' />

Author: Added a small section about Tabs.Tab

>   <Tabs.TabList>
>     <Tabs.Tab value='apple' label='Apple' key='apple' />
>     <Tabs.Tab value='orange' label='Orange' key='orange' />
>     <Tabs.Tab value='pear' label='Pear' key='pear' />
>   </Tabs.TabList>
>   <Tabs.Panel value='apple' key='apple'>
>     {/*...Content for apple*/}
>   </Tabs.Panel>
>   <Tabs.Panel value='orange' key='orange'>
>     {/*...Content for orange*/}
>   </Tabs.Panel>
>   <Tabs.Panel value='pear' key='pear'>
>     {/*...Content for pear*/}
>   </Tabs.Panel>

Reviewer 2: nit: Would be easier to understand the DOM hierarchy with new lines.

```suggestion
  <Tabs.TabList>
    <Tabs.Tab value='apple' label='Apple' key='apple' />
    <Tabs.Tab value='orange' label='Orange' key='orange' />
    <Tabs.Tab value='pear' label='Pear' key='pear' />
  </Tabs.TabList>
  
  <Tabs.Panel value='apple' key='apple'>
    {/*...Content for apple*/}
  </Tabs.Panel>
  <Tabs.Panel value='orange' key='orange'>
    {/*...Content for orange*/}
  </Tabs.Panel>
  <Tabs.Panel value='pear' key='pear'>
    {/*...Content for pear*/}
  </Tabs.Panel>
```

Same for the other code markdown below.

Author: Updated

import PropsTable from '~/components/PropsTable.astro';
import LiveExample from '~/components/LiveExample.astro';
import * as AllExamples from 'examples';

<p>{frontmatter.description}</p>

<LiveExample src='Tabs.main.tsx'>
  <AllExamples.TabsMainExample client:load />
</LiveExample>

> ### Composition API

Author: Changed docs to have this structure. Still working on words.

> ### Composition API

Author: Decided to keep composition and props section for easier routing (eg. one person wants to send a link to composition api)

> ### Composition API

Reviewer: Can we rename "Props API" to "Legacy API" then?

> ### Composition API

Author: Renamed

You can build Tabs with subcomponents that are fully customizable.

We provide:

- `Tabs.Wrapper`
- `Tabs.List`
- `Tabs.Tab`
- `Tabs.TabIcon`
- `Tabs.TabLabel`

> - `Tabs.TabDescription`

Reviewer: we should explain the convenience props in `Tabs.Tab` (e.g. `label`) and the subcomponents api

Author: Added a small section

- `Tabs.Actions`

> - `Tabs.Panel`

Author: Using this one!

> `Tabs.Tab` and `Tabs.Panel` can be associated with each other by passing them the same `value`.

Reviewer: We should also explain the `value` on `onValueChange` props (uncontrolled vs controlled) somewhere on this page

Author: Added controlled state section

### Props API

You can use Tabs directly with labels, actions, and children as panels. Although this pattern can suffice for simple cases, we strongly recommend using composition API for the best experience.

A list of all available props can be found at the end of this [page](#props).

> <LiveExample src='Tabs.legacy.tsx'>

Reviewer: if you go with the brief static code example for composition api, you can replace this live example with equivalent static code for legacy api. it will flow nicely with wording like this:

> The following code is equivalent to the composition example above:
> ```tsx
> <Tabs
>   labels={[
>     <Tab key='apple' label='Apple' />,
>     <Tab key='orange' label='Orange' />,
>     <Tab key='pear' label='Pear' />,
>   ]}
> >
>   {/* … content for active tab */}
> </Tabs>
> ```

Author: Looks good!

  <AllExamples.TabsLegacyExample client:load />
</LiveExample>

## Variants

> ### Horizontal

Author: Removed.

<LiveExample src='Tabs.horizontal.tsx'>
  <AllExamples.TabsHorizontalExample client:load />
</LiveExample>

### Vertical

<LiveExample src='Tabs.vertical.tsx'>
  <AllExamples.TabsVerticalExample client:load />
</LiveExample>

### Borderless

> As their name implies, borderless tabs have no frame or stroke surrounding them. The label appears on top of the content, above a stroke that separates it from the text area. The stroke cuts off beneath the active tab, so it can be visually associated with its respective content.

Author: Removed.

<LiveExample src='Tabs.borderless.tsx'>
  <AllExamples.TabsBorderlessExample client:load />
</LiveExample>

### Pill

Pill tabs are meant to be used with content that belong to a similar category. They act like a filter for toggling through different states of similar content.

<LiveExample src='Tabs.pill.tsx'>
  <AllExamples.TabsPillExample client:load />
</LiveExample>

### Actions

You can add action buttons to the end of tab list.

<LiveExample src='Tabs.actions.tsx'>
  <AllExamples.TabsActionsExample client:load />
</LiveExample>

> ### Overflow

Reviewer: looks like the "Vertical" example also unnecessarily scrolls the page.

https://github.com/iTwin/iTwinUI/assets/9084735/07cb58bc-29aa-40e0-8226-b72b9af9cdbd

---

the problem is likely here:

https://github.com/iTwin/iTwinUI/blob/bc6ea5b1347bd77f8dfd019e6962c4e96bad53c8/packages/itwinui-react/src/core/Tabs/Tabs.tsx#L224-L238

we could change the component to instead use `scroll-margin` in CSS

> ### Overflow

Reviewer 2: I noticed this when using the new Tabs in the migration tool too

> ### Overflow

Reviewer: will be fixed in #1639

If the available space does not allow all the tabs to be displayed, tabs become scrollable.

<LiveExample src='Tabs.overflow.tsx'>
  <AllExamples.TabsOverflowExample client:load />
</LiveExample>

> ## Usage

Author: Removed this usage in favour of other usage section.

- Use [borderless tabs](#borderless) when the tabbed content can be easily told apart from its surroundings (or lack thereof).
- The maximum recommended amount of [pill tabs](#pill) is 4.

## Props

> <PropsTable path={frontmatter.propsPath} />

Author: Not sure how to expose props for composition components.

> <PropsTable path={frontmatter.propsPath} />

Author: Added unfinished to props section
